src/App.jsx

```diff
@@ -14,45 +16,88 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_ALPHABET = 'alphabetically';
+const SORT_FIELD_LENGTH = 'length';
+
+function getPreparedGoods(goods, sortField, isReverse) {
+  const copyGoods = [...goods];
+
+  if (sortField) {
+    copyGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_ALPHABET:
+          return good1.localeCompare(good2);
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReverse) {
+    copyGoods.reverse();
+  }
+
+  return copyGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverse, setReverse] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, sortField, reverse);
+  const activeBtn = 'button is-success';
+  const notActiveBtn = 'button is-success is-light';
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`${sortField !== SORT_FIELD_ALPHABET ? notActiveBtn : activeBtn}`}
+          onClick={() => setSortField(SORT_FIELD_ALPHABET)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={`${sortField !== SORT_FIELD_LENGTH ? notActiveBtn : activeBtn}`}
+          onClick={() => setSortField(SORT_FIELD_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={`${!reverse ? notActiveBtn : activeBtn}`}
+          onClick={() => {
+            if (!reverse) {
+              setReverse(true);
+            } else {
+              setReverse(false);
+            }
+          }}
```

Comment:
move to a handler function
```
const handleReverseClick = () => {
  setIsReversed(prev => !prev)
}
```

```suggestion
          onClick={handleReverseClick}
```